### PR TITLE
Add convergence docs

### PR DESCRIPTION
**Pull request by wwilsman**
[Rendered](https://github.com/thefrontside/bigtest/blob/ww/convergence-docs/packages/convergence/README.md)

### CODE REVIEW

File: packages/convergence/README.md [hunk 1]
```diff
@@ -39,3 +39,336 @@ that condition has been met.
...
+  stats.elapsed  // time taken to converge on the entire stack
+  stats.runs     // number of times functions in the stack ran
+  stats.timeout  // this Convergence instance's timeout
+  stats.value    // value returned from the last function in the stack
+  stats.stack    // an array of stats objects from the stack
+});
+```
+
+Because `.run()` returns a promise, and `Convergence` is immutable,
+you can call `.run()` multiple times on the same convergence instance.
```
Comment (wwilsman) [inline packages/convergence/README.md]: I feel like this README ends kind of abruptly. Maybe some section to tie it all up? Or in the future we could link to other `@bigtest/*` packages that use `@bigtest/convergence`. Thoughts?

File: packages/convergence/README.md [hunk 2]
```diff
@@ -39,3 +39,336 @@ that condition has been met.
 And it isn't just for assertions either. Because it is a general
 mechanism for synchronizing on any observed state, It can be used to
 properly time test setup and teardown as well.
+
+## Creating a Convergence
+
+A convergent function is simply a function that is ran
```
Comment (cowboyd) [inline packages/convergence/README.md]: ~simply~

Comment (cowboyd) [reply packages/convergence/README.md]: s/ran/run

File: packages/convergence/README.md [hunk 3]
```diff
@@ -39,3 +39,336 @@ that condition has been met.
...
+repeatedly until it no longer throws an error. When the function is
```
Comment (cowboyd) [inline packages/convergence/README.md]: > as long as it does not return false or throw an error
right? A single false will kill it.

**Comment by wwilsman** [comment]
@cowboyd Updated based on feedback. I removed references to `convergeOn` from the `Convergence` docs. But I left a bit about the low-level notice for `convergeOn` as a sort of footnote to the rest of it.

**Pull request merged**